xcodeproj/internal/xcschemes/xcscheme_infos.bzl

```
@@ -0,0 +1,684 @@
+"""Module for parsing macro custom Xcode schemes json in the analysis phase."""
+
+load(
+    "//xcodeproj/internal:memory_efficiency.bzl",
+    "EMPTY_STRING",
+    "FALSE_ARG",
+    "TRUE_ARG",
+)
+
+# Constructors
+
+def _make_arg_env(value, *, enabled = TRUE_ARG):
+    return struct(
+        enabled = enabled,
+        value = value,
+    )
+
+def _make_build_target(id, *, post_actions = [], pre_actions = []):
+    return struct(
+        id = id,
+        post_actions = post_actions,
+        pre_actions = pre_actions,
+    )
+
+def _make_diagnostics(
+        *,
+        address_sanitizer = FALSE_ARG,
+        thread_sanitizer = FALSE_ARG,
+        undefined_behavior_sanitizer = FALSE_ARG):
+    return struct(
+        address_sanitizer = address_sanitizer,
+        thread_sanitizer = thread_sanitizer,
+        undefined_behavior_sanitizer = undefined_behavior_sanitizer,
+    )
+
+def _make_launch_target(
+        id = EMPTY_STRING,
+        *,
+        extension_host = EMPTY_STRING,
+        post_actions = [],
+        pre_actions = [],
```

Handful of default parameters where this could be done:
```suggestion
        post_actions = EMPTY_LIST,
        pre_actions = EMPTY_LIST,
```

---

I believe I didn't use that here because these variables never make it into a depset or provider. I'll have to confirm (I'm probably wrong, and it's also at the level where it _could_ so it's probably best to just do it anyway).

---

Sounds good. Up to you, just figured I'd keep calling it out but it's easy to update now or later so no worries. 👍

---

So, these are serialized to JSON. I'm not sure if they need to be memory efficient, but I'll assume so, and adjust the values in this file.